orchagent/routeorchmpls.cpp

```diff
@@ -0,0 +1,755 @@
+                    if (nextHops.contains(nexthop))
+                    {
+                        return false;
+                    }
+                }
+
+                /* Add a temporary route when a next hop group cannot be added,
+                 * and there is no temporary route right now or the current temporary
+                 * route is not pointing to a member of the next hop group to sync. */
+                addTempLabelRoute(ctx, nextHops);
+                /* Return false since the original route is not successfully added */
+                return false;
+            }
+        }
+
+        next_hop_id = m_syncdNextHopGroups[nextHops].next_hop_group_id;
+    }
+
+    /* Sync the inseg entry */
+    sai_inseg_entry_t inseg_entry;
+    // route_entry.vr_id = vrf_id; No VRF support for MPLS?
+    inseg_entry.switch_id = gSwitchId;
+    inseg_entry.label = label;
+
+    sai_attribute_t inseg_attr;
+    auto& object_statuses = ctx.object_statuses;
+
+    /* If the label is not in m_syncdLabelRoutes, then we need to create the route
+     * for this prefix with the new next hop (group) id. If the prefix is already
+     * in m_syncdLabelRoutes, then we need to update the route with a new next hop
+     * (group) id. The old next hop (group) is then not used and the reference
+     * count will decrease by 1.
+     */
+    if (it_route == m_syncdLabelRoutes.at(vrf_id).end())
+    {
+        vector<sai_attribute_t> inseg_attrs;
+        inseg_attr.id = SAI_INSEG_ENTRY_ATTR_NEXT_HOP_ID;
+        inseg_attr.value.oid = next_hop_id;
+        inseg_attrs.push_back(inseg_attr);
+        inseg_attr.id = SAI_INSEG_ENTRY_ATTR_NUM_OF_POP;
+        inseg_attr.value.u32 = ctx.pop_count;
+        inseg_attrs.push_back(inseg_attr);
+
+        /* Default SAI_INSEG_ENTRY_ATTR_PACKET_ACTION is SAI_PACKET_ACTION_FORWARD */
+        object_statuses.emplace_back();
+        sai_status_t status = gLabelRouteBulker.create_entry(&object_statuses.back(), &inseg_entry, (uint32_t)inseg_attrs.size(), inseg_attrs.data());
+        if (status == SAI_STATUS_ITEM_ALREADY_EXISTS)
+        {
+            SWSS_LOG_ERROR("Failed to create label route %u with next hop(s) %s",
+                           label, nextHops.to_string().c_str());
+            return false;
+        }
+    }
+    else
+    {
+        /* Set the packet action to forward when there was no next hop (dropped) */
+        if (it_route->second.getSize() == 0)
+        {
+            vector<sai_attribute_t> inseg_attrs;
+            inseg_attr.id = SAI_INSEG_ENTRY_ATTR_NEXT_HOP_ID;
+            inseg_attr.value.oid = next_hop_id;
+            inseg_attrs.push_back(inseg_attr);
+            inseg_attr.id = SAI_INSEG_ENTRY_ATTR_NUM_OF_POP;
+            inseg_attr.value.u32 = ctx.pop_count;
+            inseg_attrs.push_back(inseg_attr);
+
+            inseg_attr.id = SAI_INSEG_ENTRY_ATTR_PACKET_ACTION;
+            inseg_attr.value.s32 = SAI_PACKET_ACTION_FORWARD;
```

Comment:
can you please explain what is the idea and what are we doing  here ? looks like we can have two set action here ? can't we make one set action and return where we don't other attribute ?

Reply:
@abdosi This looks like something went wrong with merge when FgNhg support was added earlier this year.  I will fix this logic.

Reply:
@abdosi this logic is fixed.